modal/_container_entrypoint.py

```diff
@@ -861,6 +848,26 @@ def main(container_args: api_pb2.ContainerArguments, client: Client):
             dep_object_ids: list[str] = [dep.object_id for dep in container_args.function_def.object_dependencies]
             container_app.hydrate_function_deps(imp_fun.function, dep_object_ids)
 
+        # Identify all "enter" methods and split into groups that should run pre/post checkpointing
```

Comment:
This feels messy and I would like to refactor some of the code in `_partial_functions` to push the logic over there but it wasn't trivial so I wanted to make progress on testing basic functionality.

Reply:
Refactored by splitting the enum to have `ENTER_PRE_CHECKPOINT` and `ENTER_POST_CHECKPOINT` members. This feels less clean than the previous approach but in lieu of a larger refactor of the code that uses these flags, it ends up being a lot simpler.